### PR TITLE
Complex Class Added 

### PR DESCRIPTION
**Pull request by darkcoderrises**
Complex class was added (in sympy/sets/fancyset.py)
A test was added (in sympy/core/tests/test_argv.py)

@hargup Please see and suggest me what more to do.

@hargup writes:
- [x] Add printer: see https://github.com/sympy/sympy/pull/8902 for an example
- [x] Add rules for intersection and union with any real set.
- [ ] Squash the commits.

### CODE REVIEW

**Comment by darkcoderrises** [comment]
Can you please explain "Don't you want to assume these are reals?"

**Comment by hargup** [comment]
> Can you please explain "Don't you want to assume these are reals?"

Variables in SymPy are complex by default, you have to add `real=True` to assume them as reals.
`x = Symbol('x', real=True)`

**Comment by darkcoderrises** [comment]
okay and is the indentation around * correct?

**Comment by hargup** [comment]
- [ ] Add printer: see https://github.com/sympy/sympy/pull/8902 for an example
- [ ] Add rules for intersection and union with any real set.
- [ ] Squash the commits.

**Comment by darkcoderrises** [comment]
What rules should i add for union with any real set?

**Comment by darkcoderrises** [comment]
And are these changes right for printers, sorry but i sent that commit without testing just for asking and confirming  somethings so it may have some faults.??

**Comment by smichr** [comment]
Could you please change the title of the PR? It's not very useful to have an issue number that can't be clicked there, and the number itself doesn't tell the reviewer what they are going to be seeing. (And when there are lots of Issue ### titles it's hard to remember what's what.)

:eyeglasses: See the section **Writing pull request title and description** at [the wiki](https://github.com/sympy/sympy/wiki/Development-workflow)

**Comment by hargup** [comment]
> What rules should i add for union with any real set?

For example interestion of any the complex set with any real set is the real set and the union of the complex set with any real set is the complex set.

**Comment by darkcoderrises** [comment]
i was thinking that like i have to write rules for intersection and union of real and complex, wont there be rules for integer and reals. But i was not able to find them, can you suggest me where to look?

File: sympy/sets/fancysets.py [hunk 1]
```diff
@@ -298,6 +298,16 @@ def TransformationSet(*args, **kwargs):
     return ImageSet(*args, **kwargs)
 
 
+class Complex(with_metaclass(Singleton, ImageSet)):
```
Comment (jksuom) [inline sympy/sets/fancysets.py]: Would it be possible to think of a name more suitable for a *set* (cf. "Reals"). (The name "Complex" might have other uses in the future.)

Comment (darkcoderrises) [inline sympy/sets/fancysets.py]: what name should i give, ComplexNum?

Comment (aktech) [inline sympy/sets/fancysets.py]: How about ComplexDisk ?

Comment (jksuom) [inline sympy/sets/fancysets.py]: Or perhaps ComplexPlane. (I believe the idea was to have a set for geometric purposes.)

Comment (hargup) [inline sympy/sets/fancysets.py]: +1 for ComplexPlane.

**Comment by hargup** [comment]
> wont there be rules for integer and reals.

Then add those rules.

**Comment by darkcoderrises** [comment]
@hargup Can you tell me where should i add them(and i was talking in general, just giving an example; so i think i am not able to find them). Can you tell me in how should i proceed if i am struck like this again?

**Comment by darkcoderrises** [comment]
In [36]: a=Interval(2,2)

In [37]: b=Interval(1,1)

In [38]: a
Out[38]: {2}

In [39]: b
Out[39]: {1}

In [40]: Union(a,b)
Out[40]: {1, 2}

In [41]: c=Union(a,b)

In [42]: d= Interval(2.5,3.5)

In [43]: d
Out[43]: [2.5, 3.5]

In [44]: c
Out[44]: {1, 2}

In [45]: Union(c,d)
/usr/local/lib/python2.7/dist-packages/IPython/core/formatters.py:239: FormatterWarning: Exception in text/plain formatter: 'only unicode'
  FormatterWarning,
Out[45]: ---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-45-a9da7ff154e4> in <module>()
----> 1 Union(c,d)

/usr/local/lib/python2.7/dist-packages/IPython/core/displayhook.pyc in __call__(self, result)
    251             self.write_output_prompt()
    252             format_dict, md_dict = self.compute_format_data(result)
--> 253             self.write_format_data(format_dict, md_dict)
    254             self.update_user_ns(result)
    255             self.log_output(format_dict)

/usr/local/lib/python2.7/dist-packages/IPython/core/displayhook.pyc in write_format_data(self, format_dict, md_dict)
    172         # newline, even if all the prompt separators are ''. This is the
    173         # standard IPython behavior.
--> 174         result_repr = format_dict['text/plain']
    175         if '\n' in result_repr:
    176             # So that multi-line strings line up with the left column of

KeyError: 'text/plain'

**Comment by darkcoderrises** [comment]
I was thinking about adding _contains also please see and tell me how to make it better
    def _intersect(self, other):
        if other is S.Complex:
            return self
        elif other in S.Reals:
            return other
        return None

    def _contains(self, other):
        if re(other) in S.Reals and im(other) in S.Reals:
            return True
        elif re(other) in S.Reals is False and im(other) in S.Reals is False:
            return False

**Comment by hargup** [comment]
Looks fine, add the changes + tests and commit. Also resolve the merge conficts.

**Comment by hargup** [comment]
Closing in favour of #9463, @darkcoderrises you are invited to take part in the discuss and collaborate with @aktech if you still interested in working on Complex sets.

**Pull request closed without merging**